Revise “Logging Architecture” topic (#33876)
kubernetes/website: content/en/docs/concepts/cluster-administration/logging.md

When you run [`kubectl logs`](/docs/reference/generated/kubectl/kubectl-commands#logs) as in
the basic logging example, the kubelet on the node handles the request and
reads directly from the log file. The kubelet returns the content of the log file.

{{< note >}}
Only the contents of the latest log file are available through
`kubectl logs`.

For example, if a Pod writes 40 MiB of logs and the kubelet rotates logs
after 10 MiB, running `kubectl logs` returns at most 10MiB of data.
{{< /note >}}

## System component logs

There are two types of system components: those that run in a container and those
that do not run in a container. For example:

* The Kubernetes scheduler and kube-proxy run in a container.
* The kubelet and container runtime do not run in containers.

On machines with systemd, the kubelet and container runtime write to journald.
If systemd is not present, the kubelet and container runtime write to `.log` files in the
`/var/log` directory.
System components inside containers always write to the `/var/log` directory,
bypassing the default logging mechanism.

{{< note >}}

> If you deploy Kubernetes cluster components (such as the scheduler) to log to
> a volume shared from the parent node, you need to consider and ensure that those
> logs are rotated. Kubernetes does not manage that log rotation. Your operating
> system may automatically implement some log rotation - for example, if you share
> the directory `/var/log` into a static Pod for a component, node-level log rotation
> treats a file in that directory the same as a file written by any component outside
> Kubernetes.

Reviewer: Use [simple and direct language](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
```suggestion
Kubernetes does not manage log rotation. If you deploy Kubernetes cluster components (such as the scheduler) to log to
a volume shared from the parent node, make sure that those
logs are rotated. 

Your operating system may automatically implement some log rotation. 
For example, if you share the directory `/var/log` into a static Pod for a component, 
node-level log rotation treats a file in that directory the same as a file written
by any component outside Kubernetes.
```

Author: You know what, CRIContainerLogRotation is GA so I think this needs a bigger change.

Author: I think that's sorted, now.

by the kubelet.
{{< /note >}}

Here are two example manifests that you can use to implement a sidecar container with a logging agent.
The first manifest contains a [`ConfigMap`](/docs/tasks/configure-pod-container/configure-pod-configmap/)
to configure fluentd.

{{< codenew file="admin/logging/fluentd-sidecar-config.yaml" >}}

{{< note >}}
In the sample configurations, you can replace fluentd with any logging agent, reading
from any source inside an application container.
{{< /note >}}

The second manifest describes a pod that has a sidecar container running fluentd.
The pod mounts a volume where fluentd can pick up its configuration data.

{{< codenew file="admin/logging/two-files-counter-pod-agent-sidecar.yaml" >}}

### Exposing logs directly from the application

![Exposing logs directly from the application](/images/docs/user-guide/logging/logging-from-application.png)

> Cluster-logging that exposes or pushes logs directly from every application is outside the scope of Kubernetes.

Reviewer 2: ```suggestion
Cluster logging that exposes or pushes logs directly from every application is outside the scope of Kubernetes.
```

## {{% heading "whatsnext" %}}

* Read about [Kubernetes system logs](/docs/concepts/cluster-administration/system-logs/)
* Learn about [Traces For Kubernetes System Components](/docs/concepts/cluster-administration/system-traces/)

> * Learn how to [customise the termination message](/docs/tasks/debug/debug-application/determine-reason-pod-failure/#customizing-the-termination-message) that Kubernetes records when a Pod fails

Reviewer 2: ```suggestion
* Learn how to [customize the termination message](/docs/tasks/debug/debug-application/determine-reason-pod-failure/#customizing-the-termination-message) that Kubernetes records when a Pod fails
```

> * Learn how to a Pod that fails can [customise the termination message](/docs/tasks/debug/debug-application/determine-reason-pod-failure/#customizing-the-termination-message) that Kubernetes records

Reviewer 3: wdyt?
```suggestion
* Learn how [customise the termination message](/docs/tasks/debug/debug-application/determine-reason-pod-failure/#customizing-the-termination-message) that Kubernetes records when a Pod fails
```
